### ISSUE TITLE
Support Webpack 4

### ISSUE THREAD

**Issue by thangman22** [mozilla-neutrino/webpack-chain#45]
I try to test webpack-chain with Webpack 4 one error is not valid because of Webpack 4 is no need config anymore. So not passing any config Webpack is still work. Can we remove this test?

```javascript
test('validate empty', t => {
  const config = new Config();
  const errors = validateSchema(config.toConfig());
  t.is(errors.length, 1);
});
```

**Comment by thangman22** [comment]
After run test all test is pass only this test is fail. It should work if we upgrade to webpack 4

**Comment by edmorley** [comment]
A diff of the webpack options schema changes is here:
https://gist.github.com/edmorley/b6fd5122ed030236e25bad80621b5e76

(Generated using: `diff -U 999 <( curl -sSfL https://github.com/webpack/webpack/raw/v3.11.0/schemas/webpa ckOptionsSchema.json ) <( curl -sSfL https://github.com/webpack/webpack/raw/v4.1.1/schemas/WebpackO ptions.json ) > webpack-schema.diff`)

Of note:
* `entry` no longer required (as above)
* new `mode` option
* new `optimization` options
* `sideEffects` property for `modules.rules` entries

**Comment by eliperelman** [comment]
Fixed by #51.
